doc/specifications/proposed/rest_resource_embedding.md

```diff
@@ -0,0 +1,185 @@
+# REST resource embedding
+
+Rest embedding allows an API consumer to request references from the
+response to be embedded, in order to avoid extra REST calls.
+
+## Example
+
+A location response contains a reference to the content item's main location:
+
+```
+curl -X GET http://localhost:8000/api/ezp/v2/content/objects/1
+```
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<Content media-type="application/vnd.ez.api.ContentInfo+xml" href="/api/ezp/v2/content/objects/1" remoteId="9459d3c29e15006e45197295722c7ade" id="1">
+ <ContentType media-type="application/vnd.ez.api.ContentType+xml" href="/api/ezp/v2/content/types/1"/>
+ <Name>eZ Platform</Name>
+ <Versions media-type="application/vnd.ez.api.VersionList+xml" href="/api/ezp/v2/content/objects/1/versions"/>
+ <CurrentVersion media-type="application/vnd.ez.api.Version+xml" href="/api/ezp/v2/content/objects/1/currentversion"/>
+ <Section media-type="application/vnd.ez.api.Section+xml" href="/api/ezp/v2/content/sections/1"/>
+ <MainLocation media-type="application/vnd.ez.api.Location+xml" href="/api/ezp/v2/content/locations/1/2"/>
+ <Locations media-type="application/vnd.ez.api.LocationList+xml" href="/api/ezp/v2/content/objects/1/locations"/>
+ <Owner media-type="application/vnd.ez.api.User+xml" href="/api/ezp/v2/user/users/14"/>
+ <lastModificationDate>2015-11-30T13:10:46+00:00</lastModificationDate>
+ <publishedDate>2015-11-30T13:10:46+00:00</publishedDate>
+ <mainLanguageCode>eng-GB</mainLanguageCode>
+ <currentVersionNo>9</currentVersionNo>
+ <alwaysAvailable>true</alwaysAvailable>
+ <ObjectStates media-type="application/vnd.ez.api.ContentObjectStates+xml" href="/api/ezp/v2/content/objects/1/objectstates"/>
+</Content>
+```
+
+By adding an `X-eZ-Embed-Value` header to the request, we can get the
+main location object embedded into the response:
+
+```
+curl -X GET http://localhost:8000/api/ezp/v2/content/objects/1 -H 'x-ez-embed-value: Content.MainLocation'
```

Comment:
Would it be possible to instead use query param and align with JSON API? Or are there some implementation details that makes it hard to use query param?

As in: 
```
curl -X GET http://localhost:8000/api/ezp/v2/content/objects/1?include=Content.MainLocation
```

Reply:
(using header will make it hard to cache for instance)